src/PDecor.cc

```diff
@@ -420,6 +420,21 @@ PDecor::resize(uint width, uint height)
     renderBorder();
 }
 
+void
+PDecor::moveResize(const Geometry &geometry, int gm_mask)
+{
+    if (! gm_mask) {
+        return;
+    }
+    _gm = geometry;
```

Comment:
This causes some funky semantics, setting geometry (should be gm as _gm) .x to something but only setting Y_VALUE mask will still cause the move to happen on the X axis.

Only update the parts of _gm that is specified in the mask, should cause less surprises.